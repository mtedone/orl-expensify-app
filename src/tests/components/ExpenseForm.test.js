import React from 'react';
import { shallow } from 'enzyme';
import ExpenseForm from '../../components/ExpenseForm';
import expenses from '../fixtures/expenses';

test('should render ExpenseForm correctly', () => {
  const wrapper = shallow(<ExpenseForm />);
  expect(wrapper.debug()).toMatchSnapshot();
});

test('should render Expense Form with expense data', () => {
  const wrapper = shallow(<ExpenseForm expense={expenses[0]} />);
  expect(wrapper.debug()).toMatchSnapshot();
});

test('should handle error for invalid form submission', () => {
  const wrapper = shallow(<ExpenseForm />);
  expect(wrapper.debug()).toMatchSnapshot();
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {},
  });
  expect(wrapper.state('error').length).toBeGreaterThan(0);
  expect(wrapper.debug()).toMatchSnapshot();
});

test('should set description on input change', () => {
  const value = 'New description';
  const wrapper = shallow(<ExpenseForm />);
  wrapper
    .find('input')
    .at(0)
    .simulate('change', {
      target: { value: value },
    });
  expect(wrapper.state('description')).toBe(value);
});

test('should set note on textarea change', () => {
  const value = 'Some note';
  const wrapper = shallow(<ExpenseForm />);
  wrapper.find('textarea').simulate('change', {
    target: {
      value: value,
    },
  });
  expect(wrapper.state('note')).toBe(value);
});

test('should set amount if valid input', () => {
  const value = '12.50';
  const wrapper = shallow(<ExpenseForm />);
  wrapper
    .find('input')
    .at(1)
    .simulate('change', {
      target: {
        value: value,
      },
    });
  expect(wrapper.state('amount')).toBe(value);
});

test('should ', () => {
  const value = '12.500';
  const wrapper = shallow(<ExpenseForm />);
  const wrapperState = wrapper.state();
  wrapper
    .find('input')
    .at(1)
    .simulate('change', {
      target: {
        value: value,
      },
    });
  expect(wrapper.state()).toEqual(wrapperState);
});
