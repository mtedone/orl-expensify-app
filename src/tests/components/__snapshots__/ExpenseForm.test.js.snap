// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle error for invalid form submission 1`] = `
"<div>
  <form onSubmit={[Function]}>
    <input type=\\"text\\" placeholder=\\"Description\\" autoFocus={true} value=\\"\\" onChange={[Function]} />
    <input type=\\"text\\" placeholder=\\"Amount\\" value=\\"\\" onChange={[Function]} />
    <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} displayFormat=\\"DD MMM YYYY\\" numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"fa;lfdjal;sfjal;kdfjl;j\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} dayPickerNavigationInlineStyles={{...}} navPosition=\\"navPositionTop\\" navPrev={{...}} navNext={{...}} renderNavPrevButton={{...}} renderNavNextButton={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderWeekHeaderElement={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
    <textarea placeholder=\\"Add a note for your expense (optional)\\" value=\\"\\" onChange={[Function]} />
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;

exports[`should handle error for invalid form submission 2`] = `
"<div>
  <p>
    Please provide description and amount
  </p>
  <form onSubmit={[Function]}>
    <input type=\\"text\\" placeholder=\\"Description\\" autoFocus={true} value=\\"\\" onChange={[Function]} />
    <input type=\\"text\\" placeholder=\\"Amount\\" value=\\"\\" onChange={[Function]} />
    <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} displayFormat=\\"DD MMM YYYY\\" numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"fa;lfdjal;sfjal;kdfjl;j\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} dayPickerNavigationInlineStyles={{...}} navPosition=\\"navPositionTop\\" navPrev={{...}} navNext={{...}} renderNavPrevButton={{...}} renderNavNextButton={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderWeekHeaderElement={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
    <textarea placeholder=\\"Add a note for your expense (optional)\\" value=\\"\\" onChange={[Function]} />
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;

exports[`should render Expense Form with expense data 1`] = `
"<div>
  <form onSubmit={[Function]}>
    <input type=\\"text\\" placeholder=\\"Description\\" autoFocus={true} value=\\"Gum\\" onChange={[Function]} />
    <input type=\\"text\\" placeholder=\\"Amount\\" value=\\"2\\" onChange={[Function]} />
    <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} displayFormat=\\"DD MMM YYYY\\" numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"fa;lfdjal;sfjal;kdfjl;j\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} dayPickerNavigationInlineStyles={{...}} navPosition=\\"navPositionTop\\" navPrev={{...}} navNext={{...}} renderNavPrevButton={{...}} renderNavNextButton={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderWeekHeaderElement={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
    <textarea placeholder=\\"Add a note for your expense (optional)\\" value=\\"\\" onChange={[Function]} />
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;

exports[`should render ExpenseForm correctly 1`] = `
"<div>
  <form onSubmit={[Function]}>
    <input type=\\"text\\" placeholder=\\"Description\\" autoFocus={true} value=\\"\\" onChange={[Function]} />
    <input type=\\"text\\" placeholder=\\"Amount\\" value=\\"\\" onChange={[Function]} />
    <withStyles(SingleDatePicker) date={{...}} onDateChange={[Function]} focused={false} onFocusChange={[Function]} displayFormat=\\"DD MMM YYYY\\" numberOfMonths={1} isOutsideRange={[Function: isOutsideRange]} id=\\"fa;lfdjal;sfjal;kdfjl;j\\" placeholder=\\"Date\\" ariaLabel={[undefined]} disabled={false} required={false} readOnly={false} screenReaderInputMessage=\\"\\" showClearDate={false} showDefaultInputIcon={false} inputIconPosition=\\"before\\" customInputIcon={{...}} customCloseIcon={{...}} noBorder={false} block={false} small={false} regular={false} verticalSpacing={22} keepFocusOnInput={false} orientation=\\"horizontal\\" anchorDirection=\\"left\\" openDirection=\\"down\\" horizontalMargin={0} withPortal={false} withFullScreenPortal={false} appendToBody={false} disableScroll={false} initialVisibleMonth={{...}} firstDayOfWeek={{...}} keepOpenOnDateSelect={false} reopenPickerOnClearDate={false} renderCalendarInfo={{...}} calendarInfoPosition=\\"bottom\\" hideKeyboardShortcutsPanel={false} daySize={39} isRTL={false} verticalHeight={{...}} transitionDuration={[undefined]} horizontalMonthPadding={13} dayPickerNavigationInlineStyles={{...}} navPosition=\\"navPositionTop\\" navPrev={{...}} navNext={{...}} renderNavPrevButton={{...}} renderNavNextButton={{...}} onPrevMonthClick={[Function: onPrevMonthClick]} onNextMonthClick={[Function: onNextMonthClick]} onClose={[Function: onClose]} renderMonthText={{...}} renderWeekHeaderElement={{...}} renderCalendarDay={[undefined]} renderDayContents={{...}} renderMonthElement={{...}} enableOutsideDays={false} isDayBlocked={[Function: isDayBlocked]} isDayHighlighted={[Function: isDayHighlighted]} monthFormat=\\"MMMM YYYY\\" weekDayFormat=\\"dd\\" phrases={{...}} dayAriaLabelFormat={[undefined]} />
    <textarea placeholder=\\"Add a note for your expense (optional)\\" value=\\"\\" onChange={[Function]} />
    <button>
      Add Expense
    </button>
  </form>
</div>"
`;
